src/plugindata.h

```diff
@@ -248,6 +237,19 @@ GeanyData;
 
 #define geany			geany_data	/**< Simple macro for @c geany_data that reduces typing. */
 
+typedef struct _GeanyPluginFuncs GeanyPluginFuncs;
+
+/** Basic information for the plugin and identification.
+ * @see geany_plugin. */
+typedef struct GeanyPlugin
+{
+	PluginInfo	*info;	/**< Fields set in plugin_set_info(). */
+	GeanyData	*geany_data;	/**< Pointer to global GeanyData intance */
+	GeanyPluginFuncs *callbacks;	/**< Callbacks implemented by the plugin, set in geany_load_module() */
```

Comment:
I find it odd to have GeanyPlugin*Funcs* but the field named *callbacks*.  And as this is a virtual API, I'd rather name the field `funcs` or something (well, I probably would not have inlined the member of this struct, but I agree there is some slight advantages at doing it this way, so okay)